purchases/src/main/kotlin/com/revenuecat/purchases/galaxy/StoreProductConversions.kt

```diff
@@ -0,0 +1,114 @@
+package com.revenuecat.purchases.galaxy
+
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.models.Period
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.models.StoreProduct
+import com.samsung.android.sdk.iap.lib.vo.ProductVo
+
+internal fun ProductVo.toStoreProduct(): StoreProduct {
+    val type = this.type.createRevenueCatProductTypeFromSamsungIAPTypeString()
+    val period: Period? = if (type == ProductType.SUBS) {
+        this.createPeriod()
```

Comment:
Should we fail here if the period couldn't be parsed for a subscription product? Seems that shouldn't happen, so it would be good to fail (gracefully) in case it does 😅

Reply:
Log added here: 2449973da8d16411cad4f04f33e8956e3d4ebb95 👍